typescript/sdk/src/token/CosmosNativeWarpModule.ts

```diff
@@ -0,0 +1,469 @@
+
+    const { destinationGas: actualDestinationGas } = actualConfig;
+    const { destinationGas: expectedDestinationGas } = expectedConfig;
+    const { remoteRouters: expectedRemoteRouters } = expectedConfig;
+
+    // refetch after routes have been previously enrolled without the "actualConfig"
+    // updating
+    const { remote_routers: actualRemoteRouters } =
+      await this.signer.query.warp.RemoteRouters({
+        id: this.args.addresses.deployedTokenRoute,
+      });
+
+    const alreadyEnrolledDomains = actualRemoteRouters.map(
+      (router) => router.receiver_domain,
+    );
+
+    if (!deepEquals(actualDestinationGas, expectedDestinationGas)) {
+      // Convert { 1: 2, 2: 3, ... } to [{ 1: 2 }, { 2: 3 }]
+      const gasRouterConfigs: { domain: string; gas: string }[] = [];
+      objMap(expectedDestinationGas, (domain: string, gas: string) => {
+        gasRouterConfigs.push({
+          domain,
+          gas,
+        });
+      });
+
+      // in cosmos updating the gas config is done by unenrolling the router and then
+      // enrolling it with the updating value again
+      gasRouterConfigs.forEach(({ domain, gas }) => {
+        if (alreadyEnrolledDomains.includes(parseInt(domain))) {
+          updateTransactions.push({
+            annotation: `Unenrolling ${this.args.addresses.deployedTokenRoute} on ${this.args.chain}`,
+            typeUrl: R.MsgUnrollRemoteRouter.proto.type,
+            value: R.MsgUnrollRemoteRouter.proto.converter.create({
+              owner: actualConfig.owner,
+              token_id: this.args.addresses.deployedTokenRoute,
+              receiver_domain: parseInt(domain),
+            }),
+          });
+        }
+
+        updateTransactions.push({
+          annotation: `Setting destination gas for ${this.args.addresses.deployedTokenRoute} on ${this.args.chain}`,
+          typeUrl: R.MsgEnrollRemoteRouter.proto.type,
+          value: R.MsgEnrollRemoteRouter.proto.converter.create({
+            owner: actualConfig.owner,
+            token_id: this.args.addresses.deployedTokenRoute,
+            remote_router: {
+              receiver_domain: parseInt(domain),
+              receiver_contract: addressToBytes32(
+                expectedRemoteRouters[domain].address,
+              ),
+              gas,
+            },
+          }),
+        });
+      });
+    }
+
+    return updateTransactions;
+  }
+
+  /**
+   * Create transactions to update an existing ISM config, or deploy a new ISM and return a tx to setInterchainSecurityModule
+   *
+   * @param actualConfig - The on-chain router configuration, including the ISM configuration, and address.
+   * @param expectedConfig - The expected token router configuration, including the ISM configuration.
+   * @returns Cosmos transaction that need to be executed to update the ISM configuration.
+   */
+  async createIsmUpdateTxs(
+    actualConfig: HypTokenRouterConfig,
```

Comment:
why don't you use the `DerivedTokenRouterConfig` type here so you don't have to type cast? (applies also for other places)